webrtc/videoroom/lib/sfu/sfu.ex

```diff
@@ -0,0 +1,359 @@
+    video_track = if relay_video, do: [Track.new(:video, stream_id)], else: []
+    audio_track ++ video_track
+  end
+
+  defp get_outbound_tracks(endpoints, true) do
+    Enum.flat_map(endpoints, fn {_id, endpoint} -> Endpoint.get_tracks(endpoint) end)
+  end
+
+  defp get_outbound_tracks(_endpoints, false), do: []
+
+  defp create_links(_receive_media = true, new_endpoint_bin_name, ctx, state) do
+    flat_map_children(ctx, fn
+      {:tee, {endpoint_id, track_id}} = tee ->
+        endpoint = state.endpoints[endpoint_id]
+        track = Endpoint.get_track_by_id(endpoint, track_id)
+
+        [
+          link(tee)
+          |> via_out(:copy)
+          |> via_in(Pad.ref(:input, track_id), options: [encoding: track.encoding])
+          |> to(new_endpoint_bin_name)
+        ]
+
+      _child ->
+        []
+    end)
+  end
+
+  defp create_links(_receive_media = false, _endpoint, _ctx, _state) do
+    []
+  end
+
+  defp setup_extensions(encoding, extension_options) do
+    if encoding == :OPUS and extension_options.vad, do: [{:vad, Membrane.RTP.VAD}], else: []
+  end
+
+  defp remove_peer(peer_id, ctx, state) do
+    case do_remove_peer(peer_id, ctx, state) do
+      {:absent, [], state} ->
+        Membrane.Logger.info("Peer #{inspect(peer_id)} already removed")
+        {[], state}
+
+      {:present, actions, state} ->
+        dispatch({:peer_left, peer_id})
+        {actions, state}
+    end
+  end
+
+  defp do_remove_peer(peer_id, ctx, state) do
+    if !Map.has_key?(state.endpoints, peer_id) do
+      {:absent, [], state}
+    else
+      {endpoint, state} = pop_in(state, [:endpoints, peer_id])
+      tracks = Enum.map(Endpoint.get_tracks(endpoint), &%Track{&1 | enabled?: false})
+
+      tracks_msgs = update_track_messages(ctx, tracks, {:endpoint, peer_id}, state)
+
+      endpoint_bin = ctx.children[{:endpoint, peer_id}]
+
+      actions =
+        if endpoint_bin == nil or endpoint_bin.terminating? do
+          []
+        else
+          children =
+            Endpoint.get_tracks(endpoint)
+            |> Enum.map(fn track -> track.id end)
+            |> Enum.flat_map(&[tee: {peer_id, &1}, fake: {peer_id, &1}])
+            |> Enum.filter(&Map.has_key?(ctx.children, &1))
+
+          children = [endpoint: peer_id] ++ children
+          [remove_child: children]
+        end
+
+      {:present, tracks_msgs ++ actions, state}
+    end
+  end
+
+  defp update_track_messages(_ctx, [] = _tracks, _endpoint_bin, _state), do: []
+
+  defp update_track_messages(ctx, tracks, endpoint_bin, state) do
+    flat_map_children(ctx, fn
+      {:endpoint, other_peer_pid} = other_endpoint_bin
+      when other_endpoint_bin != endpoint_bin and other_peer_pid != state.active_screensharing ->
+        [forward: {other_endpoint_bin, {:add_tracks, tracks}}]
+
+      _child ->
+        []
+    end)
+  end
+
+  defp flat_map_children(ctx, fun) do
+    ctx.children |> Map.keys() |> Enum.flat_map(fun)
+  end
```

Comment:
I would consider deleting this function, in my opinion using it makes code less readable